### DISCUSSION TITLE
i18n support

### DISCUSSION THREAD

**Discussion by verlanmar** [arthelokyo/astrowind#49] [Ideas]
Hi!

Would love to see the internationalization support, please.

Example implementation in Astro:

https://docs.astro.build

https://github.com/withastro/astro/tree/main/examples/docs

Thank you!

**Comment by prototypa** [comment]
Hi @verlanmar 

Thank you very much for contacting us. Internationalization (i18n) is pending in **AstroWind**. Give us a few days to work on it or if you come up with a solution don't hesitate to create a PR. 

In the meantime you can read some ideas we gave to someone who needed i18n on a blog using **AstroWind**.

https://github.com/onwidget/astrowind/issues/18

**Comment by Dublinscope** [comment]
Hi @prototypa ,

I launched last month a website in German, Dutch, Swedish and Danish last month based on your template.
I gave you credit in the footer section ("Template made by [onWidget](https://onwidget.com/)":
https://roody.eu/

You can change the language with a dropdown in the navbar (top left corner).
There is nothing groundbreaking.

Please let me know whether you have any comment.

Thanks.

**Comment by prototypa** [comment]
Hi @Dublinscope 

There are fundamental aspects related to SEO (Search Engine Optimization) that it is important to take into account when making an internationalizable website. In this url you can have more details:

https://developers.google.com/search/docs/specialty/international/managing-multi-regional-sites

---

In the case of your website, I notice three details in which if you want to improve web positioning, it would be good if you updated.

1. The main **html** tag should have the _"lang"_ attribute with the appropriate language used in pages, currently it always has _"en"_

```html
<!DOCTYPE html>
<html lang="en" class="motion-safe:scroll-smooth 2xl:text-[24px]">
<head>
```

2. Tell Google about localized versions of your page (https://developers.google.com/search/docs/specialty/international/localized-versions)

```html
<head>
  <link rel="canonical" href="https://roody.eu/" />
 
  <link rel="alternate" hreflang="de" href="https://roody.eu/" />
  <link rel="alternate" hreflang="nl" href="https://roody.eu/nl/" />
  <link rel="alternate" hreflang="dk" href="https://roody.eu/dk/" />
  <link rel="alternate" hreflang="no" href="https://roody.eu/no/" />
  <link rel="alternate" hreflang="se" href="https://roody.eu/se/" />
</head>
```

3. Blog is not internationalized and have demo data that can be indexes by google:

https://roody.eu/blog/

**Comment by Dublinscope** [comment]
Hi @prototypa ,

Thank you for your comment regarding SEO.
Setting the "lang" attribute was in my to-do list.
I am going to include the SEO link tags as well.
I removed the blog pages as they are not used.
